### ISSUE TITLE
AOT compilation stability

### ISSUE THREAD

**Issue by selfsame** [arcadia-unity/Arcadia#184]
- https://github.com/arcadia-unity/Arcadia/blob/develop/Source/arcadia/internal/map_utils.clj#L1 needs to require clojure.core.protocols to succeed at `arcadia.compiler/aot-namespace`ing  `arcadia.core` in one go.
- `arcadia.config` option to `:disable-build-compile` (useful when you've manually aot'd your project, granular ignores per lib name might be nice down the road)
- compile dependency caching - right now every user `clj` file will compile it's full dep graph. Assuming that full de-duplication is not easy, perhaps the `arcadia` and `clojure` subsets could compile earlier, to be be skipped when walking user deps?
- sniff namspace meta - avoiding compilation of `(ns ^:editor foo)` could help prevent editor utilities breaking the build process - currently these kinds of namespaces have to be commented out or renamed when building.

**Comment by selfsame** [comment]
for the last two, something like the following in `arcadia.internal.editor-interop` would work well.  I also believe you don't need assume an `aot` of `clojure.core` and `arcadia.core`, just a single user namespace pass.

```clj
(defn aot-user-namespaces! [out] 
  (spit "Assets/dag.clj"
    (str (list 'ns 'dag
      (->> (all-user-namespaces-symbols)
           (remove (comp :editor meta find-ns))
           (remove #{'dag})
           (cons :require)))))
  (arcadia.compiler/aot-namespace out 'dag))
```

**Comment by timsgardner** [comment]
presumed closed as of a6b8fa7c2a23a731d2d2b01a66b1276ccdc984bc
